libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts

```diff
@@ -191,22 +192,32 @@ export class ResourceExplorer {
             throw new Error(`Resource ${typeof resourceOrId === 'string' ? resourceOrId : resourceOrId.id} not found.`);
         }
 
-        const json = resource.readText();
-        const result = JSON.parse(json, (_key: string, value: unknown): unknown => {
-            if (typeof value !== 'object') {
-                return value;
-            }
-            if (Array.isArray(value)) {
-                return value;
-            }
-            const kind = value['$kind'];
-            if (!kind) {
-                return value;
-            }
-            return this.load(value as { $kind: string } & Record<string, unknown>);
-        }) as T;
+        if (this._cache.has(resource.id)) {
+            return this._cache.get(resource.id) as T;
+        }
 
+        const json = resource.readText();
         const config = JSON.parse(json);
+
+        const result = this.preload<T>(config.$kind, resource.id);
```

Comment:
Does it make sense to ensure `$kind` is defined, and perhaps to throw an error if that isn't the case?

Reply:
I placed a throw in the `preload` method:
```
const type = this._kindToType.get(kind);
if (!type) {
    throw new Error(`Type ${kind} not registered.`);
}
``` 
Do we have to check the value of `kind` to ensure it is defined before this?

Reply:
Nope - that's good!